src/interceptors/WebSocket/index.ts

```diff
@@ -0,0 +1,31 @@
+import { BatchInterceptor } from '../..'
+import { WebSocketClient } from './WebSocketClient'
+import { WebSocketServer } from './WebSocketServer'
+import { WebSocketClassInterceptor } from './implementations/WebSocketClass/WebSocketClassInterceptor'
+
+export type WebSocketEventsMap = {
+  connection: [
+    args: {
+      /**
+       * The connected WebSocket client.
+       */
+      client: WebSocketClient
+      /**
+       * The original WebSocket server.
+       */
+      server: WebSocketServer
+    }
+  ]
+}
+
+export class WebSocketInterceptor extends BatchInterceptor<
+  [WebSocketClassInterceptor],
+  WebSocketEventsMap
+> {
+  constructor() {
+    super({
+      name: 'websocket',
+      interceptors: [new WebSocketClassInterceptor()],
```

Comment:
The `WebSocketInterceptor` itself has started as a batched interceptor on a premise that in the future it will combine multiple, transport-related interceptors (WebSocket, XHR polling). 

I believe that's redundant now. I don't plan to support any non-standard WebSocket implementations, and this pull request implements the interceptor for the standard implementation already.